pageserver/compaction/src/simulator.rs

```diff
@@ -380,8 +380,8 @@ impl interface::CompactionLayer<Key> for MockLayer {
     }
     fn file_size(&self) -> u64 {
         match self {
-            MockLayer::Delta(this) => this.file_size(),
-            MockLayer::Image(this) => this.file_size(),
```

Comment:
Collateral damage, but weirdly no error?

Reply:
it's not an error I think because it's all in the same module. I think it doesn't matter much in either direction, but probably it's better to use the functions instead of the fields. Shrug.

Reply:
well, it's not too difficult to refactor it back.

seeing how much extra LOC we get from all needless structs, getters and setters we have, I am starting to question if encapsulation attempts make any sense.